### PR TITLE
Revert "Revert "kURL allow custom vendor host preflight checks""

### PR DESCRIPTION
**Pull request by kwsorensen**
Reverts replicatedhq/kURL#2168

### CODE REVIEW

File: kurlkinds/Makefile [hunk 1]
```diff
@@ -49,6 +49,7 @@ client-gen:
 manifests:
 	controller-gen \
 		rbac:roleName=manager-role webhook crd output:crd:artifacts:config=config/crds/v1beta1 paths=./pkg/apis/...
+			sed -i 's/type: BoolString/type: string/' config/crds/v1beta1/cluster.kurl.sh_installers.yaml
```
Comment (DanStough) [inline kurlkinds/Makefile:52]: weird indent?

**Review by DanStough — COMMENTED** [review]
Any reason you opted for the separate binary vs. just building this as a subcommand in `kurl`? Seems like the number of binaries we ship is rapidly expanding.

File: kurlkinds/Makefile (hunk 1 above)
Comment (kwsorensen) [reply kurlkinds/Makefile:52]: fixed

File: scripts/Manifest [hunk 2]
```diff
@@ -9,4 +9,4 @@ KURL_UTIL_IMAGE=replicated/kurl-util:alpha
 KURL_BIN_UTILS_FILE=
 # STEP_VERSIONS array is generated by the server and injected at runtime based on supported k8s versions
 STEP_VERSIONS=("" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "1.15.3" "1.16.4" "1.17.13" "1.18.20" "1.19.12", "1.20.8")
-INSTALLER_YAML=
```
[the remark this answers is not in the document]
Comment (kwsorensen) [reply scripts/Manifest]: should be fixed

**Comment by kwsorensen** [comment]
> Any reason you opted for the separate binary vs. just building this as a subcommand in `kurl`? Seems like the number of binaries we ship is rapidly expanding.

I think that's just where @jala-dx @murphybytes and I decided to create the functionality. Not sure if we have good guidance on when to create vs integrate.

**Review by DanStough — APPROVED** [review]

**Pull request merged**